File: 005_Functions/3_prime.py
def isPrime(n):
    if n < 2:
        return False 
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False 
    return True  

def fact(a):
    if a < 0:
        return "Undefined (factorial not defined for negative numbers)"
    ans = 1
    for i in range(1, a + 1):
        ans *= i
    return ans


m = int(input("Enter num m: "))
n = int(input("Enter num n: "))

print("Is 'm' a prime number:", isPrime(m))
print("Is 'n' a prime number:", isPrime(n))
print("Is 'm-n' a prime number:", isPrime(m-n))

print("Factorial of 'm':", fact(m))
print("Factorial of 'n':", fact(n))
print("Factorial of 'm-n':", fact(m-n))

print("=========================")

# Prime Check
for num, name in [(m, 'm'), (n, 'n'), (m - n, 'm - n')]:
    if isPrime(num):
        print(f"'{name}' ({num}) is a Prime number")
    else:
        print(f"'{name}' ({num}) is NOT a Prime number")

# Factorial
for num, name in [(m, 'm'), (n, 'n'), (m - n, 'm - n')]:
    factorial_result = fact(num)
    print(f"Factorial of '{name}' ({num}): {factorial_result}")

File: 005_Functions/3_prime.cpp
#include<iostream>
using namespace std;


string Prime(int n) {
    if(n<2)
    return "No";

    int i;
    for(i=2; i<n; i++)
    {
        if(n%i==0)
        return "No";
    }
    return "Yes";
}

int Fact(int n) 
{
    int i, ans=1;
    
    for(i=1; i<=n; i++)
        ans = ans*i;
    return ans;
}

int main() {
    int a, b;
    cout<<"Enter the number: ";
    cin>>a;
    cout<<"Enter the number: ";
    cin>>b;

    cout<<"Is A a prime number? "<<Prime(a)<<endl;
    cout<<"Factorial of A: "<<Fact(a)<<endl;
    cout<<"Is B a prime number? "<<Prime(b)<<endl;
    cout<<"Factorial of B: "<<Fact(b)<<endl;
    cout<<"Is B-A a prime number? "<<Prime(b-a)<<endl;
    cout<<"Factorial of B-A: "<<Fact(b-a)<<endl;
}